pkg/machines/libvirt-dbus.js

```diff
@@ -196,6 +196,19 @@ LIBVIRT_DBUS_PROVIDER = {
         return () => call(connectionName, vmId, 'org.libvirt.Domain', 'AttachDevice', [xmlDesc, flags], TIMEOUT);
     },
 
+    CHANGE_NETWORK_AUTOSTART({
+        connectionName,
+        networkName,
+        autostart,
+    }) {
+        return (dispatch) => call(connectionName, '/org/libvirt/QEMU', 'org.libvirt.Connect', 'NetworkLookupByName', [networkName], TIMEOUT)
+                .then(networkPath => {
+                    const args = ['org.libvirt.Network', 'Autostart', cockpit.variant('b', autostart)];
+
+                    return call(connectionName, networkPath[0], 'org.freedesktop.DBus.Properties', 'Set', args, TIMEOUT);
+                });
```

Comment:
I am not sure if there is a case that it might fail. Can you please confirm with libvirt people and if there is we should add error handler here with UI error notification.

Reply:
Well, **NetworkLookupByName** API, as any other Lookup function, can fail only if it doesn't find corresponding network. _Name_ I pass into that call is derived from our list of networks, which is derived from **ListNetworks** API. So only way I can think of NetworkLookupByName could fail is that network which I'm trying to lookup was undefined between this 2 API calls, and our handler for network events didn't react in time.

For Setting DBus properties, I think only way it could fail was that network was undefined between NetworkLookupByName call and DBus-Properties-Set call. But I'm not so sure about latter one.

Right now we also don't handle failures of this sort (Lookup to get object, followed by Bus-Properties-Set) also for domains, storagepools, interfaces... So if that could be problem, we can start n discussion, issue about that and fix it for others as well.

Reply:
Ok, I am ok for this to be handled in follow up pR, since it's not failing reasonably easy. But let's do that for all aforementioned resouces.

Reply:
In follow up *

Reply:
Since all the APIs (like NetworkLookupByName in this case) call back into the daemon rather than work on top of a cache, it can very well happen that someone might have undefined a network in between calling ListNetworks and NetworkLookupByName, so one cannot assume that getting any subsequent information about a network whose name they got from the returned list will succeed too, so handling an error is desirable. In general, I think that errors should always be handled just like in the core C API unless there's a good reason to ignore failures, i.e. it's a non-fatal error which doesn't affect the overall result of the whole operation

Reply:
@eskultety  I agree. @skobyda I don't mind if you want to do that i follow up though. Just let's do that!

Reply:
I will do that in a follow up, since it needs to be changed also in other places not related to this PR